sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala

```diff
@@ -625,9 +637,21 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
                     customWebpackConfigFile match {
                       case Some(configFile) =>
                         val customConfigFileCopy = Webpack.copyCustomWebpackConfigFiles(targetDir, webpackResources.value.get)(configFile)
-                        Webpack.run("--config", customConfigFileCopy.getAbsolutePath, loader.absolutePath, bundle.absolutePath)(targetDir, logger)
+                        NpmPackage(webpackVersion).major match {
+                          case Some(4) =>
+                            // TODO: It assumes tests are run on development mode. It should instead use build settings
+                            Webpack.run("--mode", "development", "--config", customConfigFileCopy.getAbsolutePath, loader.absolutePath, "--output", bundle.absolutePath)(targetDir, logger)
+                          case _ =>
+                            Webpack.run("--config", customConfigFileCopy.getAbsolutePath, loader.absolutePath, bundle.absolutePath)(targetDir, logger)
+                        }
                       case None =>
-                        Webpack.run(loader.absolutePath, bundle.absolutePath)(targetDir, logger)
+                        NpmPackage(webpackVersion).major match {
+                          case Some(4) =>
+                            // TODO: It assumes tests are run on development mode. It should instead use build settings
+                            Webpack.run("--mode", "development", loader.absolutePath, "--output", bundle.absolutePath)(targetDir, logger)
```

Comment:
Running tests doesn’t work without `--mode development`?

Reply:
It works but you get a warning that `mode` has not been set